### DISCUSSION TITLE
Use getStaticProps in custom hook

### DISCUSSION THREAD

**Discussion by guladishvili-manual** [vercel/next.js#28827] [Ideas]
Is there any possible way to use getStaticProps in custom hook? or make it reusable for various screens? I checked react-next-hooks library but I dont want to use it since it is overkill.

Any suggestions?

**Comment by nicosh** [comment]
I dont see how you can use `getStaticProps ` (that fetch data at build time) inside an hook (the run on the client)

**Comment by pedroscursel** [comment]
I'm trying to use **getStaticProps()** ISR as it's implemented in **dev**, in a request I'll make, instead of **revalidate**. I read that the community is looking for solutions about it, does anyone here talk about it?

https://nextjs.org/docs/basic-features/data-fetching#runs-on-every-request-in-development-1
